core/extension/gdextension_interface.cpp

```diff
@@ -41,6 +41,10 @@
 #include "core/variant/variant.h"
 #include "core/version.h"
 
+#ifdef TOOLS_ENABLED
+#include "editor/editor_node.h"
+#endif
```

Comment:
`core` should not include `editor`, it should be self-contained. So this needs a different approach.

Reply:
Ok!

For the case of registering editor plugins during editor startup, this is a really simple change: we just move where the list of class names to register is stored. I only put it on `EditorPlugins` because that's where modules put them. But we can make a special static class just for GDExtensions, and `EditorNode` can reference that.

For the case of registering editor plugins when the editor is already started, it'll be a little messier, because we can't just directly call the method on `EditorNode`. I think I have an idea with passing around function pointers that will work, though.

Update coming soon! :-)

Reply:
Fixed in latest push! I added some comments by the function pointers to `EditorNode` so that future contributors know why it's doing that :-)